### PR TITLE
feat!: Rename studio service to cms

### LINKED ISSUE #877: Replace all occurances of 'studio' with 'cms'
## Context

Since [edx-platform ADR-13](https://github.com/openedx/edx-platform/blob/master/docs/decisions/0013-cms-vs-studio.rst#terminology-cms-vs-studio) was added, it's agreed the "cms" is the correct name of the webservice, whereas "studio" should only be used in a product context.

However, Devstack currently calls the service "studio". For example, to start CMS, Devstack users must run: `make dev.up.studio`; to attach to CMS, Devstack users can run: `make studio-attach`.

It may reduce confusion for Devstack users if the "studio" service (which is defined in docker-compose.yml, and referenced in docker-compose.*.yml, the Makefile, and other locations) were renamed to "cms". That would mean that users would run instead `make dev.up.cms`, `make cms-attach`, etc.

## Acceptance criteria

(this is my best guess; there may be missing steps)

* In docker-compose.yml, rename the "studio" service to "cms".
* Search the entire devstack repository for "studio". Wherever "studio" is a reference to the docker-compose service, rename it to "cms".
* Update the docs. For example, references to `make studio-shell` should be changed to `make cms-shell`.
* Make a forum announcement about the change. In the announcement, recommend that users:
  * first stop their platform: `make down`
  * then, pull new devstack code: `git pull`
  * then, confirm that they can start CMS: `make dev.up.cms`

### PR DESCRIPTION
**Pull request by UsamaSadiq**
## Description
- Resolves #877 
- Replace all occurrences of `studio` service name with `cms`. 

## Testing 
- Tested following make commands after making these changes locally: 
- `make dev.up.cms`, `make cms-shell`, `make dev.provision.lms` (it also handles provision for cms). 

## TODO [After Merging]
- [x] Communicate the change to all concerned parties.

### CODE REVIEW

File: docs/devstack_faq.rst [hunk 1]
```diff
@@ -266,31 +266,31 @@ in order to recreate up-to-date databases, static assets, etc.
 If making a patch to a named release, you should pull and use Docker images
 which were tagged for that release.
 
-Changing LMS/Studio settings
+Changing LMS/CMS settings
 ----------------------------
 
-LMS and Studio (a.k.a. CMS) read many configuration settings from the container filesystem
+LMS and CMS (a.k.a. CMS) read many configuration settings from the container filesystem
```
Comment (timmc-edx) [inline docs/devstack_faq.rst]:
```suggestion
LMS and CMS read many configuration settings from the container filesystem
```

File: docs/pycharm_integration.rst [hunk 2]
```diff
@@ -155,7 +155,7 @@ Configuration`_, with the following specific values.
 
 3. Custom run command: lms (or cms)
 
-4. Environment variables, add the following for lms/studio:
+4. Environment variables, add the following for lms/CMS:
```
Comment (timmc-edx) [inline docs/pycharm_integration.rst]: [nit] Just keeping the case the same
```suggestion
4. Environment variables, add the following for lms/cms:
```

File: provision-lms.sh [hunk 3]
```diff
@@ -80,13 +80,14 @@ done
 # Fix missing vendor file by clearing the cache
 docker-compose exec -T  lms bash -e -c 'rm /edx/app/edxapp/edx-platform/.prereqs_cache/Node_prereqs.sha1'
 
-# Create static assets for both LMS and Studio
+# Create static assets for both LMS and CMS
+# Create static assets for both LMS and CMS
```
Comment (timmc-edx) [inline provision-lms.sh]: Duplicated line

**Comment by timmc-edx** [comment]
[nit] I'd go with a `feat!` commit type here -- it's a change to the actual interface of devstack, not just its build scripts or config.

**Comment by robrap** [comment]
Note that this PR removed a communication section from the PR template: https://github.com/openedx/devstack/blob/master/.github/pull_request_template.md, which is highly applicable to this PR (I think).

**Comment by UsamaSadiq** [comment]
> Note that this PR removed a communication section from the PR template: https://github.com/openedx/devstack/blob/master/.github/pull_request_template.md, which is highly applicable to this PR (I think).

I've added a follow up TODO step to include this communication milestone in the PR.

**Review by timmc-edx — APPROVED** [review]

**Comment by UsamaSadiq** [comment]
@timmc-edx @jmbowman should we also backport this change to palm?

**Comment by timmc-edx** [comment]
Oh, probably a good idea!

**Comment by UsamaSadiq** [comment]
After discussing with Jeremy, it has been decided to delay the merging of this PR till the release of next openedx release so we won't have to do the back port and hence somehow introduce any confusion or difficulty for the community.

**Comment by UsamaSadiq** [comment]
Prioritising this PR again after the Palm release now. 
Once this PR has been merged: 

- [ ] a follow up PR will be created to update the discovery image to use `cms-url` reference instead of `studio-url` on [line](https://github.com/openedx/devstack/blob/master/provision-discovery.sh#L14C558-L14C558) and then the file `provision-discovery.sh` will be updated accordingly. Updating the file first will break the discovery provision unless the reference has been updated in the discovery image. 
- [x] A eng-wide announcement will be made for all 2U/openedx developers and community describing this change.

**Pull request merged**